src/transforms/rename.jl

```diff
@@ -0,0 +1,37 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+  Rename(:col₁ => :newcol₁, :col₂ => :newcol₂, ..., :col₁ => :newcolₙ)
+
+The transform that renames `col₁` to `newcol₁`, `col₂` to `newcol₂`, ...
+"""
+struct Rename <: Stateless
+  names::Dict{Symbol,Symbol}
+end
+
+function apply(transform::Rename, table)
```

Comment:
I think we need to `@assert` to make sure that the dictionary doesn't contain invalid oldnames. That is, before the map, we should make sure that the `keys(transform.names) \subseteq oldnames`.

Reply:
I can add an assert if you want (maybe it's a UX issue?), but right now due to the `if oldname in keys(transform.names)` invalid oldnames are gracefully ignored.

Reply:
Yes, we want to assert to alert end users that they specified incorrect column names 👍🏽